packages/studio-base/src/players/IterablePlayer/IterablePlayer.ts

```diff
@@ -932,35 +933,46 @@ export class IterablePlayer implements Player {
   }
 
   async #stateIdle() {
+    assert(this.#abort == undefined, "Invariant: some other abort controller exists");
+
     this.#isPlaying = false;
     this.#presence = PlayerPresence.PRESENT;
-    this.#queueEmitState();
 
-    if (this.#abort) {
-      throw new Error("Invariant: some other abort controller exists");
-    }
+    // set the latest value of the loaded ranges for the next emit state
+    this.#progress = {
+      fullyLoadedFractionRanges: this.#bufferedSource.loadedRanges(),
+      messageCache: this.#progress.messageCache,
+    };
+
     const abort = (this.#abort = new AbortController());
+    const aborted = new Promise((resolve) => abort.signal.addEventListener("abort", resolve));
 
-    const aborted = new Promise<void>((resolve) => {
-      abort.signal.addEventListener("abort", () => {
-        resolve();
-      });
-    });
+    // While idle, the buffered source might still be loading so we check it for range changes
+    // and queue and emit if there are changes.
+    //
+    // 100 ms second updates from the buffered source.
+    // The loadedRanges() call is memoized so this is cheap and will not queue state emit if
+    // there are no changes to the ranges.
+    const interval = setInterval(() => {
+      const newRanges = this.#bufferedSource.loadedRanges();
+      // Ranges are unchanged, so we don't need to update progress
+      if (newRanges === this.#progress.fullyLoadedFractionRanges) {
+        return;
+      }
 
-    for (;;) {
+      // Ranges changed so update progress and queue an emit
       this.#progress = {
         fullyLoadedFractionRanges: this.#bufferedSource.loadedRanges(),
         messageCache: this.#progress.messageCache,
       };
       this.#queueEmitState();
+    }, 100);
 
-      // When idling nothing is querying the source, but our buffered source might be
-      // buffering behind the scenes. Every second we emit state with an update to show that
-      // buffering is happening.
-      await Promise.race([delay(1000), aborted]);
-      if (this.#nextState) {
-        break;
-      }
+    this.#queueEmitState();
+    try {
+      await aborted;
+    } finally {
+      clearInterval(interval);
     }
```

Comment:
`aborted` never throws since you only `resolve` it
```suggestion
    await aborted;
    clearInterval(interval);
```